Rename journalist key routes

DrGFreeman opened the pull request:
## Status

Ready for review

## Description of Changes

Fixes #4421 

Changes proposed in this pull request:

Rename `why-journalist-key` and `journalist-key` routes of the Source Interface to `why-public-key` and `public-key` respectively. This aligns the route names with the language used in the associated pages where "public key" is used to designate the SecureDrop Submission Key of the instance.

Also rename the functions associated with the routes.

Maintain the `journalist-key` and redirect it to `public-key` in case it is used or linked by instance owners or sources. Add a unit test to verify the redirect.

## Testing

How should the reviewer test this PR?
Write out any special testing steps here.

1. In the source interface, navigate to `/lookup`.
1. Hover over the *public key* link.
    - [ ] Verify the link points to `/public-key`.
1. Click the *public key* link.
    - [ ] Verify the key file downloads as expected.
1. Click the *learn more* link.
    - [ ] Verify the `/why-public-key` page loads.
1. Hover over the *Download* link.
    - [ ] Verify the link points to `/public-key`.
1. Click the *Download* link.
    - [ ] Verify the key file downloads as expected.
1. Using the browser address bar navigate to `/journalist-key`.
    - [ ] Verify the redirect works and the key file downloads as expected.

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

## Checklist

### If you made changes to the server application code:

- [x] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made non-trivial code changes:

- [x] I have written a test plan and validated it for this PR

Choose one of the following:

- [x] I have opened a PR in the [docs repo](https://github.com/freedomofpress/securedrop-docs) for these changes, or will do so later
Ref.: https://github.com/freedomofpress/securedrop-docs/issues/111, https://github.com/freedomofpress/securedrop-docs/pull/112
- [ ] I would appreciate help with the documentation
- [ ] These changes do not require documentation

### If you added or updated a code dependency:

Choose one of the following:

- [ ] I have performed a diff review and pasted the contents to [the packaging wiki](https://github.com/freedomofpress/securedrop-debian-packaging/wiki)
- [ ] I would like someone else to do the diff review

DrGFreeman committed:
Rename why-journalist-key and journalist-key routes

Rename why-journalist-key and journalist-key routes of the Source
Interface to why-public-key and public-key respectively. This aligns
the route names with the language used in the associated pages where
"public key" is used to designate the SecureDrop Submission Key of the
instance.

DrGFreeman committed:
Rename functions to reflect renamed routes

emkll commented:
Thanks @DrGFreeman for your contribution. This PR will need a rebase on latest develop to include the changes to from #5648 to get all CI tests passing.

zenmonkeykstop commented on securedrop/source_app/info.py:
```diff
@@ -18,8 +18,8 @@ def tor2web_warning() -> str:
     def recommend_tor_browser() -> str:
         return render_template("use-tor-browser.html")
 
-    @view.route('/journalist-key')
```
Rather than removing this, it might be better to convert it to a permanent redirect, just on the off-chance it's being used/linked by instance owners or sources. Something like this:
```
diff --git a/securedrop/source_app/info.py b/securedrop/source_app/info.py
index df43164eb..4f1a83644 100644
--- a/securedrop/source_app/info.py
+++ b/securedrop/source_app/info.py
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
 import flask
-from flask import Blueprint, render_template, send_file, current_app
+from flask import Blueprint, render_template, send_file, current_app, redirect, url_for
 
 from io import BytesIO  # noqa
 
@@ -18,6 +18,10 @@ def make_blueprint(config: SDConfig) -> Blueprint:
     def recommend_tor_browser() -> str:
         return render_template("use-tor-browser.html")
 
+    @view.route('/journalist-key')
+    def download_journalist_key() -> flask.Response:
+            return redirect(url_for('.download_public_key'), code=301)
+    
     @view.route('/public-key')
     def download_public_key() -> flask.Response:
         journalist_pubkey = current_app.crypto_util.gpg.export_keys(
```

DrGFreeman replied:
Makes total sense!

zenmonkeykstop reviewed (changes requested):
Test plan passes, but it would be a good idea to include a redirect from the old `/journalist-key` to the new `/public-key` in case its in use anywhere. See patch in comment for simplest approach.

DrGFreeman committed:
Restore /journalist-key route and redirect to /public-key

Restore /journalist-key and redirect it to public-key in case it may be
used or linked by instances or sources.

Add a test to verify the redirect.

DrGFreeman commented:
@zenmonkeykstop, thanks for the feedback. Good points that I'll consider in future PRs.

I restored the `/journalist-key` and redirected it to `/public-key` as suggested. I also added a test to verify the redirect.

Note that the default return type of `flask.redirect` is `werkzeug.wrappers.Response` instead of `flask.Response`. It can be [changed via `Response` parameter](https://flask.palletsprojects.com/en/1.1.x/api/?highlight=redirect#flask.redirect) however I don't see the purpose of changing it at this point. Let me know if you think otherwise.

I rebased the branch as requested by @emkll. I intended to do it before submitting the PR but I forgot. 😕

zenmonkeykstop reviewed (approved):
LGTM! Thanks for adding in the 301 on the old path.
